webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/tenants/TenantService.scala

```diff
@@ -150,4 +169,76 @@ class DefaultTenantService(var tenantsEnabled: Boolean, val tenantIds: Ref[Set[T
       } yield ns.get(nodeId).filter(qc.nodePerms.canSee(_)(ts))
     }
   }
+
+  override def manageUpdate[A](
```

Comment:
:star: This is the main control logic for checking if a node can be updated by a someone, and handle cases for plugin enabled / disabled.